### PR TITLE
Navigation

### PR DESCRIPTION
**Pull request by jmalfatto**
These commits address the two navigation stories: Site and Clickable

### CODE REVIEW

**Comment by jmalfatto** [comment]
Not sure what the build errors are about, but I see they occurred yesterday and relate to module dependencies, as well. I removed my node_modules locally and did a fresh npm install without error, though it looks like I'm running an older version of node (6.9.4) than the ci server (7.4.0). Can someone pull and test this feature branch locally to confirm?

**Comment by bricemason** [comment]
@jmalfatto I pulled down your `navigation` branch and used node 7.4.0 and npm 4.0.5 via nvm. I was able to successfully complete an `npm run build` and then `npm start` to view the resulting changes.

**Comment by jmalfatto** [comment]
I tried rebuilding from circleci: with and without cache. The latter did the trick!

**Comment by jmalfatto** [comment]
Thanks for the review and the catch, @bricemason. Just pushed some style tweaks for your consideration.

File: client/components/Header/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { 
+	Route, Link
```
Comment (grgur) [inline client/components/Header/index.js]: I'd keep this in one line for consistency. When our `import` block becomes huge, it's gonna be a problem navigating through it. Just stylistic, of course

File: client/components/Header/index.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { 
+	Route, Link 
+} from 'react-router-dom';
+import './style.css';
+
+const CustomLink = ({ to, label }) => {
+	return (
+		<Route to={ to } exact children={({ match }) => <Link to={ to } className={ match.url === to ? 'selected' : ''}>{ label }</Link> } />
```
Comment (grgur) [inline client/components/Header/index.js]: This is where breaking into new lines will help

File: client/components/Header/index.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+};
+
+export default () => {
+	return (
+		<div className="header">
+			<CustomLink to="/budget" label="Budget" />
+			<CustomLink to="/reports" label="Reports" />
+		</div>
+	);
+};
```
Comment (grgur) [inline client/components/Header/index.js]: Oops, missing new line

File: client/components/Header/index.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import './style.css';
+
+const CustomLink = ({ to, label }) => {
+	return (
+		<Route to={ to } exact children={({ match }) => <Link to={ to } className={ match.url === to ? 'selected' : ''}>{ label }</Link> } />
+	);
+};
+
+export default () => {
+	return (
```
Comment (grgur) [inline client/components/Header/index.js]: no need for return.

```jsx
export default () => (
	<div className="header">
		<CustomLink to="/budget" label="Budget" />
		<CustomLink to="/reports" label="Reports" />
	</div>
);

```

File: client/components/Header/style.css [hunk 5]
```diff
@@ -0,0 +1,20 @@
+.header {
+	display: flex;
+	margin-bottom: 3em;
+}
+
+a {
```
Comment (grgur) [inline client/components/Header/style.css]: we should either move this somewhere where we keep global rules for `a` or use class name and import into this component

File: client/components/Header/style.css [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+	padding: 35px 25px;
+	margin: -6px 6px 0;
+	border-radius: 6px;
+	font-size: 16px;
+	color: #b1b1b1;
+	border: 1px solid #dadada;
```
Comment (grgur) [inline client/components/Header/style.css]: colors and various sizes are best kept in a file where we would store global variables for reuse

File: client/components/Header/style.css [hunk 7]
```diff
@@ -0,0 +1,20 @@
+.header {
```
Comment (grgur) [inline client/components/Header/style.css]: Did we say we'd keep these things in scss files?

File: client/containers/App/index.js [hunk 8]
```diff
@@ -1,8 +1,24 @@
 import React, { Component, PropTypes } from 'react';
+import { Route } from 'react-router-dom';
+import Header from 'components/Header';
 import './style.css';
 
+const Budget = (props) => {
```
Comment (grgur) [inline client/containers/App/index.js]: no need to declare props if they are not used

File: client/containers/App/index.js [hunk 9]
```diff
@@ -1,8 +1,24 @@
...
+
+const Reports = (props) => {
+  return <h1>Reports</h1>;
+};
+
 export default class App extends Component {
+
   render() {
-    return <div></div>
+    return <div>
```
Comment (grgur) [inline client/containers/App/index.js]: indentation is off here. This is easier to read:
```jsx
return (
	<div>
    		<Header />
    		<Route path="/budget" component={Budget} />
    		<Route path="/reports" component={Reports} />
   	 </div>
   );
    ```

File: client/containers/App/style.css [hunk 10]
```diff
@@ -0,0 +1,7 @@
+body {
+	box-sizing: border-box;
+	margin: 0;
+	padding: 0 3em;
+	font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif;
+	font-size: 16px;
+}
```
Comment (grgur) [inline client/containers/App/style.css]: newline

File: client/index.js [hunk 11]
```diff
@@ -1,15 +1,22 @@
...
 
 import App from 'containers/App';
 import store from 'store';
 
-import 'index.html';
-
 ReactDOM.render(
   <Provider store={store}>
-    <App />
+    <Router>
```
Comment (grgur) [inline client/index.js]: we definitely want just one component nested under Router. 
We should be able to define IndexRoute in Router config. 
Not exactly sure about React Router v4, but I hope they haven't changed that

**Comment by jmalfatto** [comment]
@grgur thanks for that code review. Based on your comments, I've applied several changes, including the addition of sass support.

all, this push requires an `npm install` to account for the new sass-loader and extract-text-webpack-plugin (the latter is recommended for production, but works just as well for development, imo).

File: client/components/Header/index.js [hunk 12]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Route, Link } from 'react-router-dom';
+import './style.scss';
```
Comment (grgur) [inline client/components/Header/index.js]: I suggest using [CSS modules](https://github.com/webpack-contrib/css-loader). The syntax will be a bit different but we get the benefit of scoping css names.

File: client/components/Header/index.js [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+const NavLink = ({ to, label }) => {
```
Comment (grgur) [inline client/components/Header/index.js]: No need for `return` :)

```jsx
const NavLink = ({ to, label }) => (
  <Route to={ to } ...
);
```

File: client/components/Header/index.js [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+		)} />
+	);
+};
+
+export default () => (
+	<div className="header">
+		<NavLink to="/budget" label="Budget" />
+		<NavLink to="/reports" label="Reports" />
+	</div>
+);
```
Comment (grgur) [inline client/components/Header/index.js]: new line

File: client/components/Header/style.scss [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+	border-radius: 6px;
+	font-size: 16px;
+	color: $gray;
+	border: 1px solid $lightgray;
+	text-decoration: none;
+
+	&:active, &.selected {
+		color: $black;
+		text-decoration: underline;
+		box-shadow: 0 1px 4px 1px $darkgray;
```
Comment (grgur) [inline client/components/Header/style.scss]: we could use this as a template shadow and `@extend` it

File: client/containers/App/_variables.scss [hunk 16]
```diff
@@ -0,0 +1,8 @@
+// fonts
```
Comment (grgur) [inline client/containers/App/_variables.scss]: Maybe put these scss files in a dedicated folder, e.g. `src/theme`?

**Comment by grgur** [comment]
eslint is erroring out on 
```
Configuration for rule "no-use-before-define" is invalid:
	Value "[object Object]" no (or more than one) schemas match.
	```

can you run lint locally?

**Comment by jmalfatto** [comment]
thanks, again, @grgur 

Now that the linter is on my radar, I'll go over this branch one more time, using and updating the lint rules as needed to get the coding style in sync with them.

As for your suggestion to use css modules, let's set up a chore for that, so as not to delay or weigh down this branch/pr any further.

**Comment by jmalfatto** [comment]
Re: the linter error, that was occurring on v2, as well. I created a separate PR on the exp/lintci branch, which addresses that issue

File: client/components/Header/style.scss [hunk 17]
```diff
@@ -0,0 +1,22 @@
+@import '~containers/App/variables';
```
Comment (grgur) [inline client/components/Header/style.scss]: I'd keep css in 2spc tabs too :)

File: package.json [hunk 18]
```diff
@@ -28,16 +28,20 @@
     "eslint-plugin-import": "2.2.0",
     "eslint-plugin-jsx-a11y": "3.0.2",
     "eslint-plugin-react": "6.9.0",
+    "extract-text-webpack-plugin": "^2.0.0",
```
Comment (grgur) [inline package.json]: let's keep dependencies to fixed versions. We don't want surprises when they update :)

**Comment by jmalfatto** [comment]
@grgur my latest commit includes the two changes I mentioned in standup: tab/space formatting and extract-text-webpack-plugin version setting. Otherwise, I think this is PR ready to go (bearing in mind the separate css chore we discussed).

**Comment by grgur** [comment]
Thanks @jmalfatto! Looks good

**Pull request merged**